objects/src/accounts/component/template/storage_entry/mod.rs

```diff
@@ -0,0 +1,715 @@
+pub use word::*;
+
+use super::AccountComponentTemplateError;
+use crate::accounts::{StorageMap, StorageSlot};
+
+mod template_key;
+pub use template_key::{TemplateKey, TemplateValue};
+
+mod init_storage_data;
+pub use init_storage_data::InitStorageData;
+
+// STORAGE ENTRY
+// ================================================================================================
+
+/// Represents a single entry in the component’s storage layout.
+///
+/// Each entry can describe:
+/// - A value slot (single word or multiple words).
+/// - A map slot (key-value map that occupies one storage slot).
+/// - A multi-slot entry (spanning multiple contiguous slots, with multiple values).
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub enum StorageEntry {
+    /// A value slot, which can contain one or more words. Each word is a hex-encoded string.
+    Value {
+        /// The human-readable name of the slot.
+        name: String,
+        /// An optional description for the slot, explaining its purpose.
+        description: Option<String>,
+        /// The numeric index of this slot in the component's storage layout.
+        slot: u8,
+        /// The initial value for this slot.
+        value: WordRepresentation,
+    },
+
+    /// A map slot, containing multiple key-value pairs. Keys and values are hex-encoded strings.
+    Map {
+        /// The human-readable name of the map slot.
+        name: String,
+        /// An optional description for the slot, explaining its purpose.
+        description: Option<String>,
+        /// The numeric index of this map slot in the component's storage.
+        slot: u8,
+        /// A list of key-value pairs to initialize in this map slot.
+        values: Vec<MapEntry>,
+    },
+
+    /// A multi-slot entry, representing a single logical value across multiple slots.
+    MultiSlot {
+        /// The human-readable name of this multi-slot entry.
+        name: String,
+        /// An optional description for the slot, explaining its purpose.
+        description: Option<String>,
+        /// The indices of the slots that form this multi-slot entry.
+        slots: Vec<u8>,
+        /// A list of values to fill the logical slot, with a length equal to the amount of slots.
+        values: Vec<WordRepresentation>,
+    },
+}
+
+impl StorageEntry {
+    /// Creates a new [`StorageEntry::Value`] variant.
+    pub fn new_value(
+        name: impl Into<String>,
+        description: Option<impl Into<String>>,
+        slot: u8,
+        value: impl Into<WordRepresentation>,
+    ) -> Self {
+        StorageEntry::Value {
+            name: name.into(),
+            description: description.map(Into::<String>::into),
+            slot,
+            value: value.into(),
+        }
+    }
+
+    /// Creates a new [`StorageEntry::Map`] variant.
+    pub fn new_map(
+        name: impl Into<String>,
+        description: Option<impl Into<String>>,
+        slot: u8,
+        storage_map: StorageMap,
+    ) -> Self {
```

Comment:
Question: why do we use `StorageMap` here rather than something like `Vec<MapEntry>`? `StorageMap` wouldn't be able to describe templated entries, right?

Reply:
Fixed this